src/dispatch/plugins/dispatch_slack/case/messages.py

```diff
@@ -321,16 +322,30 @@ def create_genai_signal_analysis_message(
     """
     signal_metadata_blocks: list[Block] = []
 
+    # we fetch the first instance id and signal
     (first_instance_id, first_instance_signal) = signal_service.get_instances_in_case(
         db_session=db_session, case_id=case.id
     ).first()
 
-    if not first_instance_id or not first_instance_signal:
-        message = "Unable to generate GenAI signal analysis. No signal instances found."
+    signal_instance = signal_service.get_signal_instance(
+        db_session=db_session, signal_instance_id=first_instance_id
+    )
+
+    # we check if the signal instance exists
+    if not signal_instance:
+        message = (
+            "Unable to generate GenAI signal analysis. No signal instances found for this case."
+        )
+        log.warning(message)
+        return create_genai_signal_message_metadata_blocks(signal_metadata_blocks, message)
```

Comment:
nit: recommend raising an exception here instead as this should really never happen outside of incorrect usage of the function

Reply:
I'll remove the check because we already check if `case.signal_instances` before going down this path.